examples/java/src/test/java/org/apache/beam/examples/complete/game/UserScoreTest.java

```diff
@@ -92,13 +104,12 @@
   /** Test the {@link ParseEventFn} {@link org.apache.beam.sdk.transforms.DoFn}. */
   @Test
```

Comment:
I'd add `@Category(ValidatesRunner.class)` here

Reply:
added

Reply:
Actually I have to issue a different comment. I didn't notice this. We save {{ValidatesRunner}} for specification tests of the Beam core. For examples, the `TestPipeline` that you use will already pick up whatever runner is configured. So you don't have to do anything for this to work as intended, running the pipeline on whatever runner is set up.

Reply:
thanks for the explanation, I've removed it

Reply:
Yes, agree with @kennknowles , sorry for misleading with that. I followed the same strategy as we have for other tests (in this test suite and in other similars) that are marked with this annotation. Don't we have to clean them up?

In the same time, `ValidatesRunner extends NeedsRunner` and Javadoc of `NeedsRunner` says:
```
/**
 * Category tag for validation tests which utilize {@link TestPipeline} for execution and expect to
 * be executed by a {@link PipelineRunner}.
 */
```
So, don't we have to mark all "TestPipeline"-ed tests with `NeedsRunner`?

Reply:
That's right. You do need to put the DirectRunner on the classpath and TestPipeline will pick it up.

The `@Category(NeedsRunner.class)` annotation is pretty much just so we can scrape them out of the SDK jar, since it cannot depend on the DirectRunner. In maven it would be a (spurious) cyclic dependency problem. With gradle I think not a problem, actually.